### DISCUSSION TITLE
Since you can't use db in `_middleware.ts`, what is the correct way of handling user auth for routes?

### DISCUSSION THREAD

**Discussion by sunneydev** [vercel/next.js#36010] [Help]
Reference: https://github.com/vercel/next.js/issues/32369

I was trying to set `user` property on my `NextRequest` using `_middleware.ts`, so I could reference them elsewhere in the API as `req.user`.

But was disappointed to find that this is not achievable as you can't use DB in the middleware.

So, what are the other possible solutions for this problem, except to get them manually for every request ?

**Comment by GabenGar** [comment]
Perform a fetch to a persistent server for those. Also I am pretty sure it's not that you "can't" use DB it's just the db drivers have to be rewritten into ESM and restructured in a way so the query functions could be imported without nodejs dependencies. Which isn't going happen any time soon.
Alternatively you can write a decorator for `getServerSideProps()` which performs auth checks and returns a callback with the `context` argument having `user` data included somewhere.
But that really depends on your usecase since the wrapper will only work on server-rendered page whereas middleware can block even static pages.

**Reply by sunneydev** [reply]
I've started using `next-connect` for this problem. But, it'd be great to have something similar in Next.js natively.
